File: rabbitmq/src/main/java/info/xiaomo/rabbitmq/config/Sender.java
package info.xiaomo.rabbitmq.config;

import org.springframework.amqp.core.AmqpTemplate;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.Date;

/**
 * @author : xiaomo
 */
@Component
public class Sender {

    private final AmqpTemplate rabbitTemplate;

    @Autowired
    public Sender(AmqpTemplate rabbitTemplate) {
        this.rabbitTemplate = rabbitTemplate;
    }

    public void send() {
        String context = "hello " + new Date();
        System.out.println("Sender : " + context);
        this.rabbitTemplate.convertAndSend("hello", context);
    }

}

File: rabbitmq/src/main/java/info/xiaomo/rabbitmq/config/Sender.kt
package info.xiaomo.rabbitmq.config

import org.springframework.amqp.core.AmqpTemplate
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import java.util.*

/**
 * @author : xiaomo
 */
@Component
class Sender @Autowired
constructor(private val rabbitTemplate: AmqpTemplate) {

    fun send() {
        val context = "hello " + Date()
        println("Sender : " + context)
        this.rabbitTemplate.convertAndSend("hello", context)
    }

}